CONTRIBUTING.md

```diff
@@ -56,34 +56,114 @@ requests - there's only so much [automated testing][ar:travis] can do.
+
+After writing the code, just before opening the pull request,
+make sure your working branch is rebased against master.
+
+### PR description
+In the description, please explain (if applicable):
+- The problem your pull request intends to solve.
+- A general overview about the implementation, or any information that can help
+reviewers to understand your code.
+- Please explain how one can try out your code: configuration snippets,
+or deployment information.
+- You can mention if you were considering alternative solutions or explain
+problems you ran into.
+- If you submit a pull request that fixes an existing issue, please mention
+the issue somewhere in the pull request, so we can close the
+original issue as well.
+
+
+The documentation is created from the description. Please provide
+a description that can be a good input for the admin guide as well.
+
+### Commits
+#### Commit messages
+
+```
+The commit messages be formatted according to this:
+
+module: short description
+
+Long description, that may be
+formatted in markdown.
+
+Signed-off-by: your name <youremail@address.com>
+```
+
+This format is checked by the CI.
+
+If you do not want to share your email address due to privacy reasons,
+you can use `yourgithubusername@users.noreply.github.com`,
+which is automatically generated and tracked by github.
+
+`module` refers to the part of syslog-ng that the patch intends to change. For example
+python, redis, persist, scratch-buffers, etc.
+
+#### Patches
+
+We are using a coding style very similar to
+[GNU Coding Standards](https://www.gnu.org/prep/standards/standards.html#Writing-C)
+for syslog-ng.
+
+You can use `make style-check` or `make style-format` to check or format automatically your code.
+These commands are executed by our CI as well.
+
+If possible, please organize your code into a set of small and self contained patches,
+with clear descriptions each, that can help to understand the patch. This greatly helps the review process.
+
+Please follow clean code guidelines whenever possible. Functions should be small and responsible for one thing.
+Try to avoid code duplication. Add descriptive names for functions and variables. Etc...
+
+#### News file
+
+We are automatically generating the news file before each release. So that to work,
+please add a news entry for your change under the news directory.
+
+The file name should be news/type-PR_ID.md (for example: news/bugfix-1234.md).
+For now the following types are supported:
+feature, bugfix, packaging, developer-note, other.
+
+If you think there is no need for a news file (for example it is a small
+fix for an earlier pull request of the same release),
+you can leave a note about it in the pull request description.
+
+The generated news file will contain a link to the pull request, where the interested
+users can find detailed information in the description.
+This means your news entry does not need to be too descriptive.
+
+The news file format should look like this:
```

Comment:
I would add something like: "See `news/README.md` for more info."

Reply:
Added.